purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1718,6 +1720,118 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
     }
 
     // endregion
+
+    // region Deprecated
+
+    /**
+     * The listener is responsible for handling changes to purchaser information.
+     * Make sure [removeUpdatedPurchaserInfoListener] is called when the listener needs to be destroyed.
+     */
+    @Deprecated(
+        "Renamed to updatedCustomerInfoListener",
+        level = DeprecationLevel.ERROR,
+        replaceWith = ReplaceWith("updatedCustomerInfoListener")
+    )
+    var updatedPurchaserInfoListener: UpdatedPurchaserInfoListener? = null
+        @JvmName("-deprecated_getUpdatedPurchaserInfoListener")
+        get
+        @JvmName("-deprecated_setUpdatedPurchaserInfoListener")
+        set
+
+    /**
+     * Get latest available purchaser info.
+     * @param listener A listener called when purchaser info is available and not stale.
+     * Called immediately if purchaser info is cached. Purchaser info can be null if an error occurred.
+     */
+    @JvmName("-deprecated_getPurchaserInfo")
+    @Deprecated(
+        "Function has been renamed to getCustomerInfo and listener has been replaced" +
+            " with ReceiveCustomerInfoListener",
+        level = DeprecationLevel.ERROR,
+        replaceWith = ReplaceWith(
+            expression = "Purchases.sharedInstance.getCustomerInfo(listener)"
```

Comment:
This is a double change: name and parameter. So it's complicated to get it to work smoothly. If changing the function first, it will change to:

```
Purchases.sharedInstance.getCustomerInfo(object: ReceivePurchaserInfoListener {
```

Which leads to a compilation error since the argument would be wrong.

Changing the parameter first would make it look like this:

```
Purchases.sharedInstance.getPurchaserInfo(object: ReceiveCustomerInfoListener {
```

Which then complains that `Purchases.sharedInstance.getPurchaserInfo` has been deprecated with an error. It also indicates that the argument is of the wrong type (it should be `ReceivePurchaserInfoListener`). So it doesn't really auto change anything. To fix that, I added another deprecated function in line 1761 for `getPurchaserInfo(listener: ReceiveCustomerInfoListener)` so it can automatically rename the whole thing.

Reply:
should we add a `getCustomerInfo(listener: ReceivePurchaserInfoListener)` version to address the first scenario?

Reply:
That makes it better thanks @beylmk !